test/rng_testsuite/extra_methods_tests/engines_methods.pass.cpp

```diff
@@ -0,0 +1,256 @@
+            (engine.max() != RANLUX24_BASE_MAX));
+}
+
+template <int N>
+std::int32_t
+check_params(oneapi::dpl::ranlux24_base_vec<N>& engine)
+{
+    return ((oneapi::dpl::ranlux24_base_vec<N>::word_size != RANLUX24_BASE_W) ||
+            (oneapi::dpl::ranlux24_base_vec<N>::short_lag != RANLUX24_BASE_S) ||
+            (oneapi::dpl::ranlux24_base_vec<N>::long_lag != RANLUX24_BASE_R) || (engine.min() != RANLUX24_BASE_MIN) ||
+            (engine.max() != RANLUX24_BASE_MAX));
+}
+
+std::int32_t
+check_params(oneapi::dpl::ranlux24& engine)
+{
+    return ((oneapi::dpl::ranlux24::block_size != RANLUX24_P) || (oneapi::dpl::ranlux24::used_block != RANLUX24_R) ||
+            (engine.min() != RANLUX24_BASE_MIN) || (engine.max() != RANLUX24_BASE_MAX));
+}
+
+template <int N>
+std::int32_t
+check_params(oneapi::dpl::ranlux24_vec<N>& engine)
+{
+    return ((oneapi::dpl::ranlux24_vec<N>::block_size != RANLUX24_P) ||
+            (oneapi::dpl::ranlux24_vec<N>::used_block != RANLUX24_R) || (engine.min() != RANLUX24_BASE_MIN) ||
+            (engine.max() != RANLUX24_BASE_MAX));
+}
+
+template <class Engine>
+bool
+test()
+{
+    using result_type = typename Engine::scalar_type;
+
+    // Catch asynchronous exceptions
+    auto exception_handler = [](sycl::exception_list exceptions) {
+        for (std::exception_ptr const& e : exceptions)
+        {
+            try
+            {
+                std::rethrow_exception(e);
+            }
+            catch (sycl::exception const& e)
+            {
+                std::cout << "Caught asynchronous SYCL exception during generation:\n" << e.what() << std::endl;
+            }
+        }
+    };
+
+    sycl::queue queue(sycl::default_selector{}, exception_handler);
+    int sum = 0;
+
+    // Memory allocation
+    std::vector<std::int32_t> dpstd_res(N_GEN);
+    constexpr std::int32_t num_elems =
+        oneapi::dpl::internal::type_traits_t<typename Engine::result_type>::num_elems == 0
+            ? 1
+            : oneapi::dpl::internal::type_traits_t<typename Engine::result_type>::num_elems;
+
+    // Random number generation
+    {
+        sycl::buffer<std::int32_t> dpstd_buffer(dpstd_res.data(), dpstd_res.size());
+
+        try
+        {
+            queue.submit([&](sycl::handler& cgh) {
+                auto dpstd_acc = dpstd_buffer.template get_access<sycl::access::mode::write>(cgh);
+
+                cgh.parallel_for<>(sycl::range<1>(N_GEN), [=](sycl::item<1> idx) {
+                    unsigned long long offset = idx.get_linear_id() * num_elems;
+                    Engine engine0(SEED);
+                    Engine engine1;
+                    engine1.seed(SEED);
+                    engine0.discard(offset);
+                    engine1.discard(offset);
+                    typename Engine::result_type res0;
+                    if constexpr (std::is_same<Engine, oneapi::dpl::ranlux24>::value)

```

Comment:
is it ok to have if constexpr here should be changed to support C++ 11

Reply:
specialization added